app/services/create_user.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: false
+
+# Creates user
+class CreateUser < ApplicationService
+  def initialize(user, cart)
+    @user = user
+    @cart = cart
+  end
+
+  def call
+    create_user
+  end
+
+  private
+
+  def create_user
+    return @cart.user unless @cart.user.nil?
+
+    user_params = @user || {}
+    User.create(**user_params.as_json.merge(guest: true))
```

Comment:
Se fizer o `user_params` ser default pra `{}` no método construtor, pode só realizar dessa forma:

```suggestion
    @cart.user || User.create(user_params.merge(guest: true))
```

Esse `as_json` pareceu desnecessário.